tiledb/sm/query/writers/global_order_writer.cc

```diff
@@ -142,6 +165,132 @@ void GlobalOrderWriter::reset() {
   }
 }
 
```

Comment:
Note to reviewers: the following additions were just moved from lower in the same file with no changes.